enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,580 @@
+* Config: a byte array that can contain a raw [nmstate](https://www.nmstate.io/) network config.
+
+nmstate is already in use within OpenShift for applying network configuration
+to nodes via the [kubernetes-nmstate
+operator](https://github.com/nmstate/kubernetes-nmstate). It uses a similar
+pattern of embedding a raw nmstate yaml structure as a byte array in the Spec.
+
+Each NMStateConfig resource will have a label that corresponds to a InstallEnv.
+The raw YAML configs for each matching resource will be rendered to a network
+config by the Assisted Service and then embedded into the discovery ISO for
+that InstallEnv. At runtime, the discovery ISO will find the config that
+matches a MAC address on the current host and then apply the config. It does
+not matter which interface has the matching MAC address; the matching is merely
+used to identify that the current host corresponds to a given config.
+
+The NMStateConfig resource design is being discussed [in a
+pull request](https://github.com/openshift/assisted-service/pull/969).
+
+#### Install Device
+
+The Agent resource Spec will include a field on which to specify the storage device
+where the OS should be installed. That field must be set by a platform-specific
+controller or some other actor that understands the underlying host.
+
+For bare metal, the BareMetalHost resource Spec already includes a
+RootDeviceHints section that will be utilized. The new Baremetal Agent
+Controller (previously described in the "Create Cluster" scenario) will use the
+BareMetalHost's RootDeviceHints and the Agent's discovery data to populate this
+field on the Agent.
+
+#### Day 2 Add Node Virtualmedia Multicluster
+
+This scenario takes place from a hub cluster, adding a worker node to a spoke cluster.
+
+1. Infra Owner creates a BareMetalHost resource with a label that matches an InstallEnv selector.
+1. The Baremetal Agent Controller adds the discovery ISO URL to the BareMetalHost.
+1. baremetal-operator uses redfish virtualmedia to boot the live ISO on the BareMetalHost.
+1. The Agent starts up and reports back to the assisted service, which creates an Agent resource in the cluster. The Agent is labeled with the labels that were specified in the InstallEnv's Spec.
+1. The Agent's Role field in its spec is assigned a value if a corresponding label and value were present on its BareMetalHost. (only "worker" is supported for now on day 2)
+1. The Agent is marked as Approved via a field in its Spec based on being recognized as running on the known BareMetalHost.
+1. The Agent runs through the validation and inspection phases. The results are shown on the Agent's Status, and eventually a condition marks the Agent as "ready".
+1. The Baremetal Agent Controller adds inspection data found on the Agent's Status to the BareMetalHost.
+1. When the agent is in a ready state, installation of that host begins.
+
+### Risks and Mitigations
```

Comment:
Should address the risk of adopting APIs now that would make it impossible in future to refactor some of these capabilities out of the assisted service into more natural components